hibernate-core/src/main/java/org/hibernate/engine/jdbc/env/internal/JdbcEnvironmentInitiator.java

```diff
@@ -239,6 +224,27 @@ else if ( useJdbcMetadata ) {
 				log.unableToObtainConnectionToQueryMetadata( e );
 			}
 		}
+		else {
+			if (someExplicitVersionSet) {
+				return new JdbcEnvironmentImpl(
+						registry,
+						dialectFactory.buildDialect(
+								configurationValues,
+								() -> new DialectResolutionInfoImpl(
+										null,
+										explicitDatabaseName,
+										explicitDatabaseVersion != null ? explicitDatabaseVersion : "0",
+										explicitDatabaseMajorVersion != null ? explicitDatabaseMajorVersion : 0,
+										explicitDatabaseMinorVersion != null ? explicitDatabaseMinorVersion : 0,
+										null,
+										0,
+										0,
+										null
```

Comment:
Not sure I understand your intention, what explicit configurations for driver etc. should i use, as compared to how it was until now?

Reply:
`org.hibernate.cfg.AvailableSettings#JAKARTA_JDBC_DRIVER` etc. should be read and passed in here

Reply:
I had wrongly assumed that we have `AvailableSettings` for configuring explicit driver metadata like driver name and version, which we otherwise determine through `DatabaseMetaData` when `useJdbcMetadata` is true. Since we don't have that, you can ignore this comment.